modern/src/runtime/GuiObject.ts

```diff
@@ -216,50 +225,87 @@ class GuiObject extends MakiObject {
     return;
   }
 
-  settargetx(x: number) {
-    unimplementedWarning("settargetx");
-    return;
+  settargetx(x: number): void {
+    this._targetParams.x = x;
   }
 
-  settargety(y: number) {
-    unimplementedWarning("settargety");
-    return;
+  settargety(y: number): void {
+    this._targetParams.y = y;
   }
 
-  settargetw(w: number) {
-    unimplementedWarning("settargetw");
-    return;
+  settargetw(w: number): void {
+    this._targetParams.w = w;
   }
 
-  settargeth(r: number) {
-    unimplementedWarning("settargeth");
-    return;
+  // r? OK...
+  settargeth(r: number): void {
+    this._targetParams.h = r;
   }
 
   // alpha range from 0-255
   settargeta(alpha: number): void {
-    unimplementedWarning("settargeta");
-    this.attributes.alpha = alpha;
-    this.js_trigger("js_update");
+    this._targetParams.alpha = alpha;
+  }
+
+  settargetspeed(insecond: number): void {
+    this._targetAnimationSpeed = insecond * 1000;
+  }
+
+  _targetAnimationLoop(): void {
+    this._targetAnimationCancelID = window.requestAnimationFrame(
+      currentTime => {
+        const progress =
+          (currentTime - this._targetAnimationStartTime) /
+          this._targetAnimationSpeed;
+        if (progress > 1) {
+          this._startParams = {};
+          this._targetParams = {};
+          this.ontargetreached();
+          return;
+        }
+        ["alpha", "x", "y", "w", "h"].forEach(attr => {
+          if (this._transitionParams[attr]) {
+            this.attributes[attr] =
+              this._targetParams[attr] * progress +
+              this._startParams[attr] * (1 - progress);
+          }
+        });
+        this.js_trigger("js_update");
+        this._targetAnimationLoop();
+      }
+    );
   }
 
-  settargetspeed(insecond: number) {
-    unimplementedWarning("settargetspeed");
-    return;
-  }
+  gototarget(): void {
+    this._transitionParams = {};
+    ["alpha", "x", "y", "w", "h"].forEach(attr => {
+      if (this._targetParams[attr] != null) {
+        this._transitionParams[attr] = true;
+        if (this.attributes[attr] != null) {
+          this._startParams[attr] = this.attributes[attr];
+        } else if (attr === "alpha") {
+          this._startParams[attr] = 255;
+        } else if (attr === "x" || attr === "y") {
+          this._startParams[attr] = 0;
+        } else {
+          this._startParams[attr] = this._targetParams[attr];
```

Comment:
This is a little funky because different attributes have different defaults. I think this will get simplified once we start adding more defaults into classes. If `x`/`y`/`alpha` all had these default values getting encoded in the constructor we could just have the first and last clause

Reply:
Actually let me just add the defaults now and simplify it hah